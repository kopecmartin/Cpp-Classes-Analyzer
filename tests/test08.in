class A {
    virtual int foo(void) =0;
};

class B {};

class C : A {
public:
    int var;
};

class D : B, protected C {
};
