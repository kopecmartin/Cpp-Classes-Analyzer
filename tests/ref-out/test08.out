<?xml version="1.0" encoding="UTF-8"?>
<model>
    <class name="A" kind="abstract">
        <private>
            <methods>
                <method name="foo" type="int" scope="instance">
                    <virtual pure="yes" />
                    <arguments></arguments>
                </method>
            </methods>
        </private> 
    </class>
    <class name="C" kind="abstract">
        <inheritance>
            <from name="A" privacy="private" />
        </inheritance>
        <public>
            <attributes>
                <attribute name="var" type="int" scope="instance" />
            </attributes>
        </public>
        <private>
            <methods>
                <method name="foo" type="int" scope="instance">
                    <from name="A" />
                    <virtual pure="yes" />
                    <arguments></arguments>
                </method>
            </methods>
        </private>
    </class>
    <class name="B" kind="concrete">
    </class>
    <class name="D" kind="abstract">
        <inheritance>
            <from name="C" privacy="protected" />
            <from name="B" privacy="private" />
        </inheritance>
        <protected>
            <attributes>
                <attribute name="var" type="int" scope="instance">
                    <from name="C" />
                </attribute>
            </attributes>
        </protected>
        <private>
            <methods>
                <method name="foo" type="int" scope="instance">
                    <from name="A" />
                    <virtual pure="yes" />
                    <arguments></arguments>
                </method>
            </methods>
        </private>
    </class>
</model>
